Sources/XiCliCore/CommandLineTool.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+public final class CommandLineTool {
+    private let arguments: [String]
+    
+    public init(arguments: [String] = CommandLine.arguments) {
+        self.arguments = Array(arguments.dropFirst())
+    }
+    
+    public func run() throws {
+        let args = try getArguments(from: arguments)
+        
+        let filePath = try resolvePath(from: args.fileInput)
+        try openFile(at: filePath)
+        
+        if args.wait {
+            print("waiting for editor to close...")
+            let group = DispatchGroup()
+            group.enter()
+            setObserver(group: group, filePath: filePath)
+            group.wait()
+            print("closed")
+        }
+
+    }
+    
+    func getArguments(from args: [String]) throws -> Arguments {
+        var fileName: String?
+        var wait = false
+        for arg in args {
+            if arg == "--wait" || arg == "-W" {
+                wait = true
+            } else {
+                fileName = arg
+            }
+        }
+        
+        guard let comfFileName = fileName else {
+            throw Error.missingFileName
+        }
+        
+        return Arguments(fileInput: comfFileName, wait: wait)
+    }
+    
+    func resolvePath(from input: String) throws -> String {
+        let fileManager = FileManager.default
+        var filePath: URL!
+        
+        if input.hasPrefix("/") {
+            filePath = URL(fileURLWithPath: input)
+        } else if input.hasPrefix("~") {
+            var input = input
+            input.remove(at: input.startIndex)
+            filePath = URL(fileURLWithPath: input)
+            print(filePath)
+        } else {
+            let basePath = URL(fileURLWithPath: fileManager.currentDirectoryPath)
+            filePath = basePath.appendingPathComponent(input)
+        }
+        
+        let pathString = filePath.path
+        guard FileManager.default.fileExists(atPath: pathString) else {
```

Comment:
I think that if the user types `xi non_existent_file.txt` we want to open a new window, referencing that path.